cpp/benchmarks/io/json/nested_json.cpp

```diff
@@ -24,15 +24,115 @@
 
 #include <tests/io/fst/common.hpp>
 
+#include <cudf/detail/utilities/vector_factories.hpp>
 #include <cudf/scalar/scalar_factories.hpp>
 #include <cudf/strings/repeat_strings.hpp>
 #include <cudf/types.hpp>
 
-#include <cstdlib>
+#include <string>
+#include <vector>
 
-namespace cudf {
 namespace {
-auto make_test_json_data(size_type string_size, rmm::cuda_stream_view stream)
+
+// pre-generate all the number strings
+std::vector<std::string> _num_to_string;
+std::string num_to_string(int32_t num) { return _num_to_string.at(num); }
+
+// List of List nested.
+std::string generate_list_of_lists(int32_t max_depth, int32_t max_rows, std::string elem)
+{
+  std::string json = "[";
+  if (max_depth > 1) json += std::string(max_depth - 1, '[');
+  for (int32_t row = 0; row < max_rows; ++row) {
+    json += elem;
+    if (row < max_rows - 1) { json += ", "; }
+  }
+  if (max_depth > 1) json += std::string(max_depth - 1, ']');
+  json += "]";
+  return json;
+}
+
+// Struct of Struct nested.
+std::string generate_struct_of_structs(int32_t max_depth, int32_t max_rows, std::string elem)
+{
+  if (max_depth <= 0) return "{}";
+  std::string json;
+  for (int32_t depth = 0; depth < max_depth / 2; ++depth) {
+    json += R"({"a)" + num_to_string(depth) + R"(": )";
+  }
+  if (max_rows == 0) json += "{}";
+
+  for (int32_t row = 0; row < max_rows; ++row) {
+    json += elem;
+    if (row < max_rows - 1) {
+      json += R"(, "a)" + num_to_string(max_depth / 2 - 1) + "_" + num_to_string(row) + R"(": )";
+    }
+  }
+  if (max_depth > 0) json += std::string(max_depth / 2, '}');
+  return json;
+}
+
+// Memoize the generated rows so we don't have to regenerate them.
+std::map<std::tuple<int, int, int, int>, std::string> _row_cache;
+
+std::string generate_row(
+  int num_columns, int max_depth, int max_list_size, int max_struct_size, size_t max_bytes)
+{
+  std::string s = "{";
+  const std::vector<std::string> elems{
+    R"(1)", R"(-2)", R"(3.4)", R"("5")", R"("abcdefghij")", R"(true)", R"(null)"};
+  for (int i = 0; i < num_columns; i++) {
+    s += R"("col)" + num_to_string(i) + R"(": )";
+    if (auto it = _row_cache.find({i % 2, max_depth - 2, max_struct_size, i % elems.size()});
+        it != _row_cache.end()) {
+      s += it->second;
+    } else {
+      if (i % 2 == 0) {
+        auto r =
+          generate_struct_of_structs(max_depth - 2, max_struct_size, elems[i % elems.size()]);
+        _row_cache[{i % 2, max_depth - 2, max_struct_size, i % elems.size()}] = r;
+        s += r;
+      } else {
+        auto r = generate_list_of_lists(max_depth - 2, max_struct_size, elems[i % elems.size()]);
+        _row_cache[{i % 2, max_depth - 2, max_struct_size, i % elems.size()}] = r;
+        s += r;
+      }
```

Comment:
or something like this
```suggestion
      auto r = (i % 2 == 0) ?
          generate_struct_of_structs(max_depth - 2, max_struct_size, elems[i % elems.size()]) : 
          generate_list_of_lists(max_depth - 2, max_struct_size, elems[i % elems.size()]);
      _row_cache[{i % 2, max_depth, max_struct_size, i % elems.size()}] = r;
      s += r;
```

Reply:
used this suggestion.